### PR TITLE
feat(monitor): Add channel monitor service for Phase 3

### PR DESCRIPTION
**Pull request by yuanying**
## Summary

- Add `ChannelMonitor` protocol in domain layer for platform-independent channel monitoring
- Implement `SlackChannelMonitor` with Slack API integration
- Add `message_limit` configuration to `ResponseConfig`

### Features

- **get_channels**: Retrieve list of channels the bot has joined
- **get_recent_messages**: Fetch recent messages with optional time filter
- **get_unreplied_messages**: Detect messages that need bot response based on:
  - Wait time threshold
  - Bot has not already replied (in channel or thread)
  - Not bot's own message

### Files Changed

| File | Change |
|------|--------|
| `src/myao2/domain/services/channel_monitor.py` | New - Protocol definition |
| `src/myao2/infrastructure/slack/channel_monitor.py` | New - Slack implementation |
| `src/myao2/config/models.py` | Modified - Add `message_limit` |
| `src/myao2/config/loader.py` | Modified - Load `message_limit` |
| `config.yaml.example` | Modified - Add `message_limit` example |
| `tests/infrastructure/slack/test_channel_monitor.py` | New - 14 test cases |

## Test plan

- [x] All 144 tests pass
- [x] Linter (ruff) passes
- [x] Type checker (ty) passes
- [x] New tests cover get_channels, get_recent_messages, get_unreplied_messages
- [x] Edge cases: API errors, bot messages, thread replies, wait time

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/myao2/infrastructure/slack/channel_monitor.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            message_ts: Message timestamp to check for replies after.
+
+        Returns:
+            True if bot has replied after the message.
+        """
+        try:
+            response = await self._client.conversations_replies(
+                channel=channel_id,
+                ts=thread_ts,
+                limit=50,
```
[the remark this answers is not in the document]
Comment (yuanying) [reply src/myao2/infrastructure/slack/channel_monitor.py]: Fixed. The thread replies limit now uses the same `self._message_limit` configuration.

File: src/myao2/infrastructure/slack/channel_monitor.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            channel_id: Channel ID.
+
+        Returns:
+            Message entity.
+        """
+        user_id = msg.get("user", "")
+        if user_id:
+            user = await self._get_user_info(user_id)
+        else:
+            user = User(id="", name="Unknown", is_bot=True)
```
[the remark this answers is not in the document]
Comment (yuanying) [reply src/myao2/infrastructure/slack/channel_monitor.py]: Fixed. Changed to use `User(id="system", name="System", is_bot=True)` with a comment explaining that these are system messages without a user field.
